Spring-Boot/Spring-Boot-Problem/src/main/java/tw/waterball/judgegirl/springboot/problem/controllers/ProblemController.java

```diff
@@ -77,6 +77,9 @@
     public List<?> getProblems(@RequestHeader(value = "Authorization", required = false) String authorization,
                                @RequestParam(value = "tags", required = false) String[] tags,
                                @RequestParam(value = "page", defaultValue = "0") int page,
+                               @RequestParam(value = "visible", defaultValue = "true") boolean visible,
+                               @RequestParam(value = "invisible", defaultValue = "true") boolean invisible,
+                               @RequestParam(value = "archive", defaultValue = "false") boolean archive,
```

Comment:
`@RequestParam(value = "archive", required = false) Boolean archive`

remember to write its getter in this way
```java
public Optional<Boolean> queryArchive()
```